### PR TITLE
test: py3 required on master except snuba

### PR DESCRIPTION
**Pull request by joshuarli**
Forgot to add this.

Also snuba migrations aren't needed in GHA, since the snuba devserver (via sentry devservices) runs them during startup.

### CODE REVIEW

**Review by billyvg — APPROVED** [review]
Should we skip `check-missing-migration` on master?

**Comment by joshuarli** [comment]
> Should we skip `check-missing-migration` on master?

Yeah, that would be a good optimization. Gimme one sec.

**Comment by joshuarli** [comment]
At some point, should add like `!contains(github.event.commits[0].message, '[skip ci]')` to all jobs (or we could namespace it like `skip-ci-backend`), it's a nice thing to have esp. for people who continuously push to a WIP PR.

**Review by billyvg — APPROVED** [review]

**Comment by joshuarli** [comment]
@billyvg I'm preemptively making this required, won't be merging until everything succeeds. Regardless, this is ready for review.

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -104,10 +108,17 @@ jobs:
...
+  # TODO(joshuarli):
+  # More jobs. Also, python2 equivalents as well.
+  # - kakfa+zookeeper devservices necessary for what test suites?
+  # - MIGRATIONS_TEST_MIGRATE=1
+  # - TEST_SUITE=relay-integration
+  # - TEST_SUITE=symbolicator
+  # - TEST_SUITE=cli
+
+  test-py3-simple-no-migrations:
+    name: '(required) python3.6 backend (simple, no migrations)'
```
Comment (billyvg) [inline .github/workflows/python.yml:120]: Is having `(required)` necessary? If we really want to make it required, it should be done at the GitHub branch enforcement level

**Review by billyvg — APPROVED** [review]

File: .github/workflows/python.yml (hunk 1 above)
Comment (joshuarli) [reply .github/workflows/python.yml:120]: Similar to the previous `[ignore fails]`, this is just here to reduce confusion for now.

**Review by BYK — APPROVED** [review]

**Comment by joshuarli** [comment]
Rebased, and also made py3 snuba + py2+py3 acceptance required, so requesting rereview.

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 name: python
 on:
+  push:
+    branches:
+      - master
```
Comment (BYK) [inline .github/workflows/python.yml:5]: Can we get `releases/*` too?

File: .github/workflows/python.yml [hunk 3]
```diff
@@ -8,6 +11,7 @@ on:
 
 jobs:
   check-missing-migration:
+    if: ${{ github.ref != 'refs/heads/master' }}
```
Comment (BYK) [inline .github/workflows/python.yml:14]: If you do the above, I guess this needs to be updated too.

**Review by BYK — COMMENTED** [review]
LGTM

File: .github/workflows/python.yml (hunk 3 above)
Comment (joshuarli) [reply .github/workflows/python.yml:14]: It's fine, not running this on master/main is the low hanging fruit.

**Comment by joshuarli** [comment]
@wedamija Modified this PR to keep snuba not required. The intention is to supersede https://github.com/getsentry/sentry/pull/21078.

File: .github/workflows/python.yml [hunk 4]
```diff
@@ -104,10 +109,9 @@ jobs:
           # Below will exit with non-zero status if model changes are missing migrations
           sentry django makemigrations -n ci_test --check --dry-run --no-input || (echo '::error::Error: Migration required -- to generate a migration, run `sentry django makemigrations -n <some_name>`' && exit 1)
 
-  test-py3:
-    name: 'py3 [ignore fails]'
+  test-py3-simple-no-migrations:
+    name: 'python3.6 backend (simple, no migrations)'
```
Comment (wedamija) [inline .github/workflows/python.yml:113]: Migrations are running on these now, so should update the name

File: .github/workflows/python.yml [hunk 5]
```diff
@@ -204,7 +201,7 @@ jobs:
           TEST_GROUP: ${{ matrix.instance }}
 
   test-py3-snuba:
-    name: 'py3 snuba [ignore fails]'
+    name: '[ignore fails] python3.6 backend (snuba)'
```
Comment (EvanPurkhiser) [inline .github/workflows/python.yml:204]: Should we change this to unicode bold `[𝗜𝗚𝗡𝗢𝗥𝗘 𝗙𝗔𝗜𝗟𝗦]` lol

**Review by wedamija — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]

File: .github/workflows/python.yml (hunk 5 above)
Comment (joshuarli) [reply .github/workflows/python.yml:204]: lol

File: .github/workflows/python.yml (hunk 4 above)
Comment (joshuarli) [reply .github/workflows/python.yml]: Good catch, I overlooked this while merging with latest.

File: .github/workflows/python.yml (hunk 3 above)
Comment (BYK) [reply .github/workflows/python.yml:15]: I really don't want this thing to be chiming in on release branches when I'm cutting a release in the future. We can do this later but I feel like it is a must, not a nice to have.

File: .github/workflows/python.yml [hunk 6]
```diff
@@ -104,10 +109,9 @@ jobs:
           # Below will exit with non-zero status if model changes are missing migrations
           sentry django makemigrations -n ci_test --check --dry-run --no-input || (echo '::error::Error: Migration required -- to generate a migration, run `sentry django makemigrations -n <some_name>`' && exit 1)
 
-  test-py3:
-    name: 'py3 [ignore fails]'
+  test-py3-simple:
+    name: 'python3.6 backend (simple)'
```
Comment (wedamija) [inline .github/workflows/python.yml:113]: What's simple? I think we should just call this the same as what we call it in travis. So `Backend` or `Backend with migrations`

Comment (joshuarli) [reply .github/workflows/python.yml:113]: 👍 then I'll figure out BYK's change and then merge

File: .github/workflows/python.yml (hunk 3 above)
Comment (joshuarli) [reply .github/workflows/python.yml:15]: nm, will just do it in this PR. https://github.com/getsentry/sentry/pull/20537#discussion_r498442637

**Review by BYK — APPROVED** [review]

**Pull request merged**